### PR TITLE
added pd df for better op viusalization

### PR DESCRIPTION
**Pull request by Prateek-Ch**
This is how the output will look like
image

### CODE REVIEW

**Comment by ZhuYuqicheng** [comment]
It seems we currently only have the triple count for the test set. I reviewed our meeting notes, and we had agreed to include triple counts for both the training set (training degree) and the test set (test degree). Could you please add the training degree to the results as well?

Additionally, in the algorithm’s output, you might notice categories like "1-N" and "M-1" that indicate the relation type. It would be great to add a "relation type" column in our table to capture this information.

**Comment by Prateek-Ch** [comment]
> It seems we currently only have the triple count for the test set. I reviewed our meeting notes, and we had agreed to include triple counts for both the training set (training degree) and the test set (test degree). Could you please add the training degree to the results as well?
>  Yes I have made this change, I completely missed this part.
> Additionally, in the algorithm’s output, you might notice categories like "1-N" and "M-1" that indicate the relation type. It would be great to add a "relation type" column in our table to capture this information.

**Comment by Prateek-Ch** [comment]
> It seems we currently only have the triple count for the test set. I reviewed our meeting notes, and we had agreed to include triple counts for both the training set (training degree) and the test set (test degree). Could you please add the training degree to the results as well?
> 
> Additionally, in the algorithm’s output, you might notice categories like "1-N" and "M-1" that indicate the relation type. It would be great to add a "relation type" column in our table to capture this information.

I have made the change that includes train triple count in the dataframe as well for better understanding of training degree. I completely missed that part by mistake.

But for the relation type, I am not sure if we have that information. The output doesn't include directly relation type data but rather in the metrics we have something like,
```
[82365f71]   hits_at_1_1-1: 
[82365f71]   hits_at_1_1-N: 
[82365f71]   hits_at_1_M-1: 
[82365f71]   hits_at_1_M-N: 
```

We have these for each and every metric that's available. So do we have to include 4 extra columns per metric as well?
The table might be too clumsy.

I am attaching  a sample output file for a single relation as well.
[output.txt](https://github.com/user-attachments/files/17651336/output.txt)

**Comment by ZhuYuqicheng** [comment]
> > It seems we currently only have the triple count for the test set. I reviewed our meeting notes, and we had agreed to include triple counts for both the training set (training degree) and the test set (test degree). Could you please add the training degree to the results as well?
> > Additionally, in the algorithm’s output, you might notice categories like "1-N" and "M-1" that indicate the relation type. It would be great to add a "relation type" column in our table to capture this information.
> 
> I have made the change that includes train triple count in the dataframe as well for better understanding of training degree. I completely missed that part by mistake.
> 
> But for the relation type, I am not sure if we have that information. The output doesn't include directly relation type data but rather in the metrics we have something like,
> 
> ```
> [82365f71]   hits_at_1_1-1: 
> [82365f71]   hits_at_1_1-N: 
> [82365f71]   hits_at_1_M-1: 
> [82365f71]   hits_at_1_M-N: 
> ```
> 
> We have these for each and every metric that's available. So do we have to include 4 extra columns per metric as well? The table might be too clumsy.
> 
> I am attaching a sample output file for a single relation as well. [output.txt](https://github.com/user-attachments/files/17651336/output.txt)

Let me make it more clear:

You can retrieve the relation types in [YOUR_KGE_MODEL].dataset.index("relations_per_type"). This is a Python dictionary where each key represents a relation type (e.g., "1-1," "1-N," "M-1," "M-N"), and the corresponding value is the list of relation IDs that fall under that type.

Please add an additional column called "relation type" as shown below:
"relation type",    "ID"
          1-1                7
          1-1                9
          1-N               8

Please let me know if you have any further question. :)

**Comment by Prateek-Ch** [comment]
> > > It seems we currently only have the triple count for the test set. I reviewed our meeting notes, and we had agreed to include triple counts for both the training set (training degree) and the test set (test degree). Could you please add the training degree to the results as well?
> > > Additionally, in the algorithm’s output, you might notice categories like "1-N" and "M-1" that indicate the relation type. It would be great to add a "relation type" column in our table to capture this information.
> > 
> > 
> > I have made the change that includes train triple count in the dataframe as well for better understanding of training degree. I completely missed that part by mistake.
> > But for the relation type, I am not sure if we have that information. The output doesn't include directly relation type data but rather in the metrics we have something like,
> > ```
> > [82365f71]   hits_at_1_1-1: 
> > [82365f71]   hits_at_1_1-N: 
> > [82365f71]   hits_at_1_M-1: 
> > [82365f71]   hits_at_1_M-N: 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > We have these for each and every metric that's available. So do we have to include 4 extra columns per metric as well? The table might be too clumsy.
> > I am attaching a sample output file for a single relation as well. [output.txt](https://github.com/user-attachments/files/17651336/output.txt)
> 
> Let me make it more clear:
> 
> You can retrieve the relation types in [YOUR_KGE_MODEL].dataset.index("relations_per_type"). This is a Python dictionary where each key represents a relation type (e.g., "1-1," "1-N," "M-1," "M-N"), and the corresponding value is the list of relation IDs that fall under that type.
> 
> Please add an additional column called "relation type" as shown below: "relation type", "ID" 1-1 7 1-1 9 1-N 8
> 
> Please let me know if you have any further question. :)

I wasn't aware of this relation type method that is available, my bad.
I have added both the requested columns and updated the above output conversation for you to have a look at the updated table.

**Comment by ZhuYuqicheng** [comment]
This looks fantastic now!

P.S. I wanted to add the relation type because it helps explain why certain relations perform well even with a small number of training triples—1-1 relations, for example, are easier to learn. :)
